crates/aptos-jwk-consensus/src/jwk_manager/mod.rs

```diff
@@ -0,0 +1,297 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    network::IncomingRpcRequest,
+    types::{JWKConsensusMsg, ObservedUpdate, ObservedUpdateResponse},
+};
+use anyhow::{anyhow, bail, Result};
+use aptos_channels::aptos_channel;
+use aptos_crypto::{bls12381::PrivateKey, SigningKey};
+use aptos_types::{
+    account_address::AccountAddress,
+    epoch_state::EpochState,
+    jwks::{Issuer, ObservedJWKs, ProviderJWKs, QuorumCertifiedUpdate, JWK},
+    validator_txn::ValidatorTransaction,
+};
+use aptos_validator_transaction_pool as vtxn_pool;
+use certified_update_producer::CertifiedUpdateProducer;
+use futures_util::future::AbortHandle;
+use std::{
+    collections::{BTreeMap, HashMap},
+    sync::Arc,
+};
+
+pub mod certified_update_producer;
+
+/// `JWKManager` executes per-issuer JWK consensus sessions
+/// and updates validator txn pool with quorum-certified JWK updates.
+pub struct JWKManager {
+    signing_key: PrivateKey,
+    my_addr: AccountAddress,
+    epoch_state: EpochState,
+    certified_update_producer: Arc<dyn CertifiedUpdateProducer>,
+    certified_update_tx: Option<aptos_channel::Sender<(), QuorumCertifiedUpdate>>,
+    vtxn_pool_write_cli: Arc<vtxn_pool::SingleTopicWriteClient>,
+    states_by_issuer: HashMap<Issuer, PerProviderState>,
```

Comment:
shall we use a String as key and do the Vec<u8> right before converting to move value?

Reply:
since issuer has been defined to be vec<u8> across all the structs, no need to convert back and forth?